clang/lib/Driver/Driver.cpp

```diff
@@ -3700,10 +3708,25 @@ class OffloadingActionBuilder final {
       // The host depends on the generated integrated header from the device
       // compilation.
       if (CurPhase == phases::Compile) {
+        bool SYCLDeviceOnly = Args.hasArg(options::OPT_fsycl_device_only);
         for (Action *&A : SYCLDeviceActions) {
           DeviceCompilerInput =
               C.MakeAction<CompileJobAction>(A, types::TY_SYCL_Header);
-          A = C.MakeAction<CompileJobAction>(A, types::TY_LLVM_BC);
+          types::ID OutputType = types::TY_LLVM_BC;
+          if (SYCLDeviceOnly) {
```

Comment:
Doesn't this block make [this one](https://github.com/intel/llvm/blob/sycl/clang/lib/Driver/Driver.cpp#L5414) obsolete? If not, do you think there's a way to remove this duplication by some additional refactoring?

Reply:
It does make that block obsolete - I even removed it in this review, so I'm a little confused... :)

Reply:
Sorry for the miss :D At least we've gotten a truly independent double-check!